cfgov/v1/atomic_elements/organisms.py

```diff
@@ -846,30 +846,114 @@ def get_context(self, value, parent_context=None):
         return context
 
 
+class VideoPlayerStructValue(blocks.StructValue):
+    @property
+    def thumbnail_url(self):
+        thumbnail_image = self.get('thumbnail_image')
+
+        if thumbnail_image:
+            return thumbnail_image.get_rendition('original').url
+
+
 class VideoPlayer(blocks.StructBlock):
-    video_url = blocks.RegexBlock(
-        label='YouTube Embed URL',
-        default='https://www.youtube.com/embed/',
-        required=True,
-        regex=r'^https:\/\/www\.youtube\.com\/embed\/.+$',
+    YOUTUBE_ID_HELP_TEXT = (
+        'Enter the YouTube video ID, which is located at the end of the video'
+        ' URL, after "v=". For example, the video ID for '
+        'https://www.youtube.com/watch?v=1V0Ax9OIc84 is 1V0Ax9OIc84.'
+    )
+
+    video_id = blocks.RegexBlock(
+        label='YouTube video ID',
+        # Set required=False to allow for non-required VideoPlayers.
+        # See https://github.com/wagtail/wagtail/issues/2665.
+        required=False,
+        # This is a reasonable but not official regex for YouTube video IDs.
+        # https://webapps.stackexchange.com/a/54448
+        regex=r'^[\w-]{11}$',
         error_messages={
-            'required': 'The YouTube URL field is required for video players.',
-            'invalid': "The YouTube URL is in the wrong format. "
-                       "You must use the embed URL "
-                       "(https://www.youtube.com/embed/video_id), "
-                       "which can be obtained by clicking Share > Embed "
-                       "on the YouTube video page.",
-        }
+            'invalid': "The YouTube video ID is in the wrong format.",
+        },
+        help_text=mark_safe(YOUTUBE_ID_HELP_TEXT)
     )
+    thumbnail_image = images_blocks.ImageChooserBlock(
+        required=False,
+        help_text=mark_safe(
+            'Optional thumbnail image to show before or after the video plays.'
+            ' If the thumbnail image is not set here, the video player will '
+            'show the thumbnail that was set in YouTube instead.'
+        )
+    )
+
+    def clean(self, value):
+        cleaned = super().clean(value)
+
+        errors = {}
+
+        if not cleaned['video_id']:
+            if getattr(self.meta, 'required', True):
+                errors['video_id'] = ErrorList([
+                    ValidationError('This field is required.'),
+                ])
```

Comment:
This is needed because of https://github.com/wagtail/wagtail/issues/2665. We want to be able to make VideoPlayers optional sub-blocks of other blocks (like FeaturedContent). But we still want `video_id` to be required anytime a VideoPlayer is defined.

We have to define `video_id` with `required=False`, otherwise you'd never be able to leave VideoPlayer blank. But if VideoPlayer is defined like `VideoPlayer(required=True)`, then in that case we do want `video_id` to act as if it is required. That's what this logic does.

Reply:
What about wrapping the VideoPlayer in a ListBlock or StreamBlockj with a max of one item when used in a place like the FCM? Then we could make `video_id` required, right?

Reply:
I suppose, but what would the advantage be? The user experience in Wagtail would be significantly worse, both visually and because the max count isn't respected on the frontend, and it'd make the block storage less obvious.

Reply:
The advantage would be being able to require `video_id` everywhere that a VideoPlayer block is added to a page, including within FCMs. It's arguable whether that UX improvement is offset by the issues you note with that approach, but given that it would also eliminate this custom clean complexity, I would argue that it's better.